crates/sui-indexer-alt-jsonrpc/src/api/transactions/mod.rs

```diff
@@ -103,9 +103,33 @@ impl QueryTransactionsApiServer for QueryTransactions {
 
         let options = query.options.unwrap_or_default();
 
-        let tx_futures = digests
-            .iter()
-            .map(|d| response::transaction(ctx, *d, &options));
+        let tx_futures = digests.iter().map(|d| {
+            let options = options.clone();
+            async move {
+                let mut tx = response::transaction(ctx, *d, &options).await;
```

Comment:
My intuition for this is not great and I usually just need to try it and see if the compiler complains, but do you need to clone `options` here, or could you avoid that by making the `async` block non-`move`? Asking because that is effectively what the previous code was doing.

```suggestion
            async {
                let mut tx = response::transaction(ctx, *d, &options).await;
```